### PR TITLE
BOAC-1036, /admin page lists all advisors and admins with a 'become' link per user

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/BOAC-1036

* this feature is not _view_as_. Instead, it gives:
-- fast access to dev_auth login
-- convenient view on `authorized_users` table  
* `dev_auth.py`, `cas_auth.py` and this new feature all rolled into `auth_controller.py`
* feature only available to admins and if DEV_AUTH is enabled

### CODE REVIEW

File: tests/test_api/test_user_controller.py [hunk 1]
```diff
@@ -87,6 +88,28 @@ def test_includes_groups(self, client, fake_auth):
...
+class TestAllUserProfiles:
+    """User Profiles API."""
+
+    def test_not_authenticated(self, client):
+        """Returns a well-formed response."""
+        response = client.get('/api/profiles/all')
+        assert response.status_code == 401
+
+    def test_not_authorized(self, client, fake_auth):
+        """Returns a well-formed response."""
```
Comment (johncrossman) [inline tests/test_api/test_user_controller.py]: This test description (and one above) needs to be amended.

File: tests/test_api/test_user_controller.py [hunk 2]
```diff
@@ -87,6 +88,28 @@ def test_includes_groups(self, client, fake_auth):
...
+class TestAllUserProfiles:
+    """User Profiles API."""
+
+    def test_not_authenticated(self, client):
+        """Returns 'unauthorized' response status if user is not authenticated."""
+        response = client.get('/api/profiles/all')
+        assert response.status_code == 401
+
+    def test_not_authorized(self, client, fake_auth):
+        """Returns 'unauthorized' response status if user is not admin."""
```
Comment (johncrossman) [inline tests/test_api/test_user_controller.py]: Proper test descriptions are in.

File: boac/api/user_controller.py [hunk 3]
```diff
@@ -72,6 +73,31 @@ def user_profile():
     return tolerant_jsonify(profile)
 
 
+@app.route('/api/profiles/all')
+@admin_required
+def all_user_profiles():
+    # This feature is not available in production
+    if app.config['DEVELOPER_AUTH_ENABLED']:
```
Comment (johncrossman) [inline boac/api/user_controller.py]: Both front-end controller and this api endpoint check for config `devAuthEnabled`.

**Comment by paulkerschen** [comment]
download

**Pull request merged**